src/lib/Components/StickyTabPage/StickyTabPageFlatList.tsx

```diff
@@ -0,0 +1,230 @@
+import { useAnimatedValue } from "./reanimatedHelpers"
+import { TAB_BAR_HEIGHT } from "./StickyTabPageTabBar"
+
+interface FlatListRequiredContext {
+  headerHeight: Animated.Node<number>
+  headerOffsetY: Animated.Value<number>
+  tabIndex: number
+  activeTabIndex: Animated.Node<number>
+}
+export const StickyTabPageFlatListContext = React.createContext<FlatListRequiredContext>(null)
+
+const MOCK_CONTEXT: () => FlatListRequiredContext = () => ({
+  headerHeight: new Animated.Value(0),
+  headerOffsetY: new Animated.Value(0),
+  tabIndex: 0,
+  activeTabIndex: new Animated.Value(0),
+})
+
+const AnimatedFlatList: typeof FlatList = Animated.createAnimatedComponent(FlatList)
+export interface StickyTabSection {
+  /* key must be unique per-tab */
+  key: string
+  content: JSX.Element
+}
+export interface StickyTabFlatListProps
+  extends Omit<FlatListProps<any>, "onScroll" | "data" | "scrollEventThrottle" | "ListHeaderComponent" | "renderItem"> {
+  data: StickyTabSection[]
+}
+
+export const StickyTabPageFlatList: React.FC<StickyTabFlatListProps> = props => {
+  const { headerHeight, headerOffsetY, tabIndex, activeTabIndex } =
+    process.env.NODE_ENV === "test" ? MOCK_CONTEXT() : useContext(StickyTabPageFlatListContext)
+
+  const contentHeight = useAnimatedValue(0)
+  const layoutHeight = useAnimatedValue(0)
+  const scrollOffsetY = useAnimatedValue(0)
+
+  const isActive = Animated.eq(tabIndex, activeTabIndex)
+
+  const { lockHeaderPosition } = useStickyHeaderPositioning({
+    headerOffsetY,
+    contentHeight,
+    layoutHeight,
+    headerHeight,
+    scrollOffsetY,
+  })
+
+  const flatListRef = useRef<{ getNode(): FlatList<any> }>()
+
+  const lastIsActive = useAnimatedValue(-1)
+
+  // prevent this tab from manipulating the header position when it is not active
+  Animated.useCode(
+    () =>
+      // when the active state changes
+      Animated.cond(Animated.neq(lastIsActive, isActive), [
+        Animated.set(lastIsActive, isActive),
+        Animated.cond(
+          isActive,
+          [
+            // the tab just became active so we might need to adjust the scroll offset to avoid unwanted
+            // white space before allowing the scroll offset to affect the header position
+            Animated.cond(
+              Animated.greaterThan(Animated.multiply(-1, headerOffsetY), scrollOffsetY),
+              Animated.call([headerOffsetY], ([y]) => {
+                if (!flatListRef.current) {
+                  throw new Error(
+                    "Please make sure that tab content is wrapped with a StickyTabPageFlatList or a StickyTabPageScrollView"
+                  )
+                }
+                flatListRef.current.getNode().scrollToOffset({ offset: -y, animated: false })
+                lockHeaderPosition.setValue(0)
+              }),
+              Animated.set(lockHeaderPosition, 0)
+            ),
+          ],
+          Animated.set(lockHeaderPosition, 1)
```

Comment:
Before this was running in JS land in a `useEffect` while now the `isActive` state is in reanimated so it has to run on the native side. Same logic though.